src/item_factory.cpp

```diff
@@ -1559,6 +1559,7 @@ void Item_factory::load( islot_magazine &slot, JsonObject &jo, const std::string
     assign( jo, "reliability", slot.reliability, strict, 0, 10 );
     assign( jo, "reload_time", slot.reload_time, strict, 0 );
     assign( jo, "linkage", slot.linkage, strict );
+    assign( jo, "compatible_guns", slot.compatible_guns, strict );
```

Comment:
We already have guns refer to magazines, is it necessary to have it go both ways?
I assume this is because of the hardcoded implication that gun with a compatible magazine has no internal magazine and gun without compatible magazine has an internal one.

It would probably be cleaner if it was like this:
* Gun gets marked as having an internal magazine by a `bool`, calculated at load time by checking if a compatible non-speedloader magazine exists (that is, non-speedloader magazine implies no internal magazine)
* Internal magazine check now returns this new `bool`
* Speedloader uses most of magazine logic as is. The only difference is that after finished loading, it doesn't get into the gun, just transfers the ammo.

Reply:
You're totally right, I've discarded this approach and I'm treating them exactly like magazines,  but with minimal fixups to allow guns with integral magazines to also use speedloaders.